src/open_dread_rando/pickups/split_pickups.py

```diff
@@ -0,0 +1,394 @@
+        return self._get_group(0)
+
+    @beam.setter
+    def beam(self, value: BillBoardGroupParams):
+        self._set_group(0, value)
+
+    @property
+    def charge(self) -> BillBoardGroupParams:
+        return self._get_group(1)
+
+    @charge.setter
+    def charge(self, value: BillBoardGroupParams):
+        self._set_group(1, value)
+
+    @property
+    def boost(self) -> BillBoardGroupParams:
+        return self._get_group(2)
+
+    @boost.setter
+    def boost(self, value: BillBoardGroupParams):
+        self._set_group(2, value)
+
+
+class SetGunChargeParams(CCFunc):
+    @property
+    def name(self) -> str:
+        return self.get_param(1)
+
+    @name.setter
+    def name(self, value: str):
+        self.set_param(1, value)
+
+
+@dataclasses.dataclass
+class PrimaryGunFuncs:
+    add_primary_gun: AddPrimaryGun
+    set_billboard_group_params: SetBillBoardGroupParams | None
+    set_gun_charge_params: SetGunChargeParams | None
+
+    @property
+    def as_list(self) -> list[dict]:
+        ret = [self.add_primary_gun.raw]
+        if self.set_billboard_group_params is not None:
+            ret.append(self.set_billboard_group_params.raw)
+        if self.set_gun_charge_params is not None:
+            ret.append(self.set_gun_charge_params.raw)
+        return ret
+
+    @property
+    def name(self) -> str:
+        return self.add_primary_gun.params.name
+
+    @name.setter
+    def name(self, value: str):
+        self.add_primary_gun.params = dataclasses.replace(
+            self.add_primary_gun.params,
+            name=value,
+        )
+        if self.set_billboard_group_params is not None:
+            self.set_billboard_group_params.name = value
+        if self.set_gun_charge_params is not None:
+            self.set_gun_charge_params.name = value
+
+
+
+class AddSecondaryGun(CCFunc):
+    pass
+
+
+SAMUS_BMSAD_PATH = "actors/characters/samus/charclasses/samus.bmsad"
+
+def patch_split_pickups(editor: PatcherEditor):
+    samus = editor.get_file(SAMUS_BMSAD_PATH, Bmsad)
+
+    primaries = _patch_split_beams(editor)
+    secondaries = _patch_split_missiles(editor)
+    selections = [
+        {
+            "name": "SelectGun",
+            "unk": 1,
+            "params": {
+                "Param1": {
+                    "type": "s",
+                    "value": "PowerBeam",
+                },
+                "Param2": {
+                    "type": "b",
+                    "value": "True"
+                }
+            }
+        },
+        {
+            "name": "SelectGun",
+            "unk": 1,
+            "params": {
+                "Param1": {
+                    "type": "s",
+                    "value": "Missile",
+                },
+                "Param2": {
+                    "type": "b",
+                    "value": "False"
+                }
+            }
+        }
+    ]
+
+    samus.raw.property.components["GUN"].functions = primaries + secondaries + selections
```

Comment:
primaries and secondaries are lists of containers. selections is a list of dicts. this makes building fail. (unless my edits were wrong.

Reply:
it shouldn't. Containers are just fancy dicts and we use them like this all over the place